src/services/GenerateVerifiableCredential.ts

```diff
@@ -0,0 +1,457 @@
+  ): any {
+  	const ci: string[] = [];
+	
+  	const addToCI = (code: string | string[]) => {
+  		if (Array.isArray(code)) {
+  			ci.push(...code);
+  		} else {
+  			ci.push(code);
+  		}
+  	};
+	
+  	const handleFaceMatchRejection = () => {
+  		const { value, reason } = ID_DOCUMENT_FACE_MATCH_RECOMMENDATION;
+  		if (value === "REJECT") {
+  			switch (reason) {
+  				case "FACE_NOT_GENUINE":
+  				case "LARGE_AGE_GAP":
+  				case "PHOTO_OF_MASK":
+  				case "PHOTO_OF_PHOTO":
+  				case "DIFFERENT_PERSON":
+  					addToCI("V01");
+  					break;
+  				default:
+  					break;
+  			}
+  		}
+  	};
+	
+  	const handleAuthenticityRejection = () => {
+  		const { value, reason } = ID_DOCUMENT_AUTHENTICITY_RECOMMENDATION;
+  		if (value === "REJECT") {
+  			switch (reason) {
+  				case "COUNTERFEIT":
+  					addToCI("D14");
+  					break;
+  				case "EXPIRED_DOCUMENT":
+  					addToCI("D16");
+  					break;
+  				case "FRAUD_LIST_MATCH":
+  					addToCI(["F03", "D14"]);
+  					break;
+  				case "DOC_NUMBER_INVALID":
+  					addToCI("D02");
+  					break;
+  				case "TAMPERED":
+  				case "DATA_MISMATCH":
+  				case "CHIP_DATA_INTEGRITY_FAILED":
+  				case "CHIP_SIGNATURE_VERIFICATION_FAILED":
+  				case "CHIP_CSCA_VERIFICATION_FAILED":
+  					addToCI("D14");
+  					break;
+  				default:
+  					break;
+  			}
+  		}
+  	};
+	
+  	handleFaceMatchRejection();
+  	handleAuthenticityRejection();
+	
+  	return ci;
+  }
+
+  private attachPersonName(
+  	credentialSubject: VerifiedCredentialSubject,
+  	givenName: string,
+  	familyName: string,
+  ): VerifiedCredentialSubject {
+  	const givenNames = givenName.split(" ");
+  	const nameParts = givenNames.map((name) => ({ value: name, type: "GivenName" }));
+	
+  	nameParts.push({ value: familyName, type: "FamilyName" });
+	
+  	credentialSubject.name = [{ nameParts }];
+	
+  	return credentialSubject;
+  }
+
+  private attachDOB(
+  	credentialSubject: VerifiedCredentialSubject,
+  	dateOfBirth: string,
+  ): VerifiedCredentialSubject {
+  	credentialSubject.birthDate = [{ value: dateOfBirth }];
+	
+  	return credentialSubject;
+  }
+	
+
+  private attachAddressInfo(
+  	credentialSubject: VerifiedCredentialSubject,
+  	postalAddress: any,
+  ): VerifiedCredentialSubject {
+  	credentialSubject.address = [
+  		{
+  			buildingNumber: postalAddress.building_number,
+  			addressLocality: postalAddress.town_city,
+  			postalCode: postalAddress.postal_code,
+  			addressCountry: postalAddress.country,
+  		},
+  	];
+	
+  	return credentialSubject;
+  }
+
+  private attachEvidencePayload(
+  	credentialSubject: VerifiedCredentialSubject,
+  	documentType: string,
+		issuingCountry: string,
+  	documentFields: any,
+  ): VerifiedCredentialSubject {
+
+		console.log('documentType', documentType);
+		console.log('issuingCountry', issuingCountry);
+		console.log('documentFields', documentFields);
+		console.log('issuing_authority', documentFields.issuing_authority);
```

Comment:
we need to use the `logger` to emit logs. Also we must not emit any PII in logs.

Reply:
Sorry these were part of failing unit test debugging, have now removed